crime-applications-adaptor/api-tests/src/test/java/uk/gov/justice/laa/crime/applications/adaptor/stepdefinition/APITestSteps.java

```diff
@@ -38,15 +53,83 @@ public void theFollowingApplicationExistsInTheDatastore(int usn) {
 
   @When("the GET internal V1 crimeapply endpoint is called with usn {int} and user {string}")
   public void theGetCrimeApplyEndpointIsCalledWith(int usn, String user) {
-    caaResponse = crimeApplicationsAdaptorAPI.getApplicationByUsn(usn, user,
-        CAA_APPLICATION_SCHEMA);
+    ValidatableResponse response = crimeApplicationsAdaptorAPI.getApplicationByUsn(usn, user);
+    response.log().all();
+
+    Serenity.setSessionVariable(USN_KEY).to(usn);
+    Serenity.setSessionVariable(RESPONSE_KEY).to(response);
   }
 
   @Then("the returned response should match the contents of {string}")
   public void theFollowingDataShouldBeReturned(String expectedResponseFile) throws JSONException {
     JsonPath expectedJson = new JsonPath(
         new File(EXPECTED_RESPONSE_FILE_PATH_BASE + expectedResponseFile));
-    JSONAssert.assertEquals(expectedJson.prettify(), caaResponse.body().asPrettyString(),
+
+    ValidatableResponse validatableResponse = (ValidatableResponse) Serenity.getCurrentSession()
+        .get(RESPONSE_KEY);
+
+    Response response = validatableResponse
+        .assertThat()
+        .statusCode(200)
+        .body(JsonSchemaValidator.matchesJsonSchemaInClasspath(CAA_APPLICATION_SCHEMA))
+        .extract()
+        .response();
+
+    JSONAssert.assertEquals(expectedJson.prettify(),
+        response.body().asPrettyString(),
         JSONCompareMode.LENIENT);
   }
+
+  @Given("an application with usn {int} does not exists in the datastore")
+  public void anApplicationWithUsnUsnDoesNotExistsInTheDatastore(int usn) {
+    crimeApplyMockAPI.getCrimeApplyMockApplicationByUsn(usn, HttpStatus.NOT_FOUND_404);
+  }
+
+  @Then("the returned response should indicate the application is not found")
+  public void theReturnedResponseShouldIndicateTheApplicationIsNotFound() {
+    ValidatableResponse validatableResponse = getSessionResponseData();
+
+    Response response = validatableResponse
+        .assertThat()
+        .statusCode(HttpStatus.NOT_FOUND_404)
+        .extract()
+        .response();
+
+    JsonPath responseJsonPath = response.jsonPath();
+    assertThat(responseJsonPath.getString("title"), equalTo("Not Found"));
+    assertThat(responseJsonPath.getString("detail"), containsString("404 Not Found from GET"));
+  }
+
+  @And("no entry for usn {int} should be present in the eforms staging table")
```

Comment:
I wonder if it might be a good idea to be specific with the table name here. If we say EFORMS_STAGING it's clear we are talking about a particular table. "eforms staging table" is a bit vague for someone unfamiliar with the setup - it could be interpreted as an unspecified table within the eforms staging environment.

Reply:
Good shout!

Reply:
I've done this for the EFORM_HISTORY table as well.